edpop_explorer/readers/fbtee.py

```diff
@@ -0,0 +1,107 @@
+            return_string += '\nAuthors:\n' + '\n'.join(authorstrings)
+        return return_string
+
+    def __repr__(self):
+        return self.get_title()
+
+
+class FBTEEReader(APIReader):
+    DATABASE_URL = 'https://dhstatic.hum.uu.nl/edpop/cl.sqlite3'
+    DATABASE_LICENSE = 'https://dhstatic.hum.uu.nl/edpop/LICENSE.txt'
+    FBTEE_LINK = 'http://fbtee.uws.edu.au/stn/interface/browse.php?t=book&' \
+        'id={}'
+
+    def __init__(self):
+        self.database_file = Path(
+            AppDirs('edpop-explorer', 'cdh').user_data_dir
+        ) / 'cl.sqlite3'
+        if not self.database_file.exists():
+            self._download_database()
+        self.con = sqlite3.connect(str(self.database_file))
+
+    def _download_database(self):
+        print('Downloading database...')
+        response = requests.get(self.DATABASE_URL)
+        if response.ok:
+            try:
+                with open(self.database_file, 'wb') as f:
+                    f.write(response.content)
+            except OSError as err:
+                raise APIException(
+                    'Error writing database file to disk: {}'.format(err)
+                )
+        else:
+            raise APIException(
+                'Error downloading database file from {}'
+                .format(self.DATABASE_URL)
+            )
+        print('Successfully saved database to {}.'.format(self.database_file))
+        print('See license: {}'.format(self.DATABASE_LICENSE))
+
+    def prepare_query(self, query: str):
+        self.prepared_query = '%' + query + '%'
+
+    def fetch(self) -> List[FBTEERecord]:
+        if not self.prepared_query:
+            raise APIException('First call prepare_query method')
+
+        cur = self.con.cursor()
+        columns = [x[1] for x in cur.execute('PRAGMA table_info(books)')]
+        res = cur.execute(
+            'SELECT B.*, BA.author_code, A.author_name FROM books B '
+            'JOIN books_authors BA on B.book_code=BA.book_code '
+            'JOIN authors A on BA.author_code=A.author_code '
+            'WHERE full_book_title LIKE ? '
+            'ORDER BY B.book_code',
+            (self.prepared_query,)
+        )
+        self.records = []
+        last_book_code = ''
+        for row in res:
+            # Since we are joining with another table, a book may be repeated,
+            # so check if this is a new item
+            book_code: str = row[columns.index('book_code')]
+            if last_book_code != book_code:
+                record = FBTEERecord()
+                record.data = {}
+                for i in range(len(columns)):
+                    record.data[columns[i]] = row[i]
+                record.link = self.FBTEE_LINK.format(book_code)
+                self.records.append(record)
+                last_book_code = book_code
+            # Add author_code and author_name to the last record
+            assert len(self.records) > 0
+            author_code = row[len(columns)]
```

Comment:
`row[len(columns)]` feels a bit brittle since it depends on column ordering. Fine in this context, but you could use a [row factory](https://docs.python.org/3.10/library/sqlite3.html#how-to-create-and-use-row-factories) to work with result rows as dicts:

```
def dict_factory(cursor, row):
    fields = [column[0] for column in cursor.description]
    return {key: value for key, value in zip(fields, row)}

con.row_factory = dict_factory
```